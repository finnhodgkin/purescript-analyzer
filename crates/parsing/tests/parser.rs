macro_rules! parser_tests {
    ($($name:ident => $source:expr),+ $(,)?) => {
        $(
            #[test]
            fn $name() {
                let lexed = lexing::lex($source);
                let tokens = lexing::layout(&lexed);
                let (node, errors) = parsing::parse(&lexed, &tokens);
                insta::assert_debug_snapshot!((node, errors));
            }
        )+
    };
}

parser_tests!(
    comment_pre_post => include_str!("parser/CommentPrePost.purs"),
    lexer_error_inserted => include_str!("parser/LexerErrorInserted.purs"),
    module_export => include_str!("parser/ModuleExport.purs"),
    module_export_class => include_str!("parser/ModuleExportClass.purs"),
    module_export_class_error_end => include_str!("parser/ModuleExportClassErrorEnd.purs"),
    module_export_comma_error => include_str!("parser/ModuleExportCommaError.purs"),
    module_export_data => include_str!("parser/ModuleExportData.purs"),
    module_export_data_comma_error => include_str!("parser/ModuleExportDataCommaError.purs"),
    module_export_data_item_after_double_period => include_str!("parser/ModuleExportDataItemAfterDoublePeriod.purs"),
    module_export_data_item_before_double_period => include_str!("parser/ModuleExportDataItemBeforeDoublePeriod.purs"),
    module_export_data_item_invalid => include_str!("parser/ModuleExportDataItemInvalid.purs"),
    module_export_empty => include_str!("parser/ModuleExportEmpty.purs"),
    module_export_invalid => include_str!("parser/ModuleExportInvalid.purs"),
    module_export_type_operator => include_str!("parser/ModuleExportTypeOperator.purs"),
    module_export_type_operator_error => include_str!("parser/ModuleExportTypeOperatorError.purs"),
    module_export_type_operator_error_end => include_str!("parser/ModuleExportTypeOperatorErrorEnd.purs"),
    module_header => include_str!("parser/ModuleHeader.purs"),
    module_header_prefixed => include_str!("parser/ModuleHeaderPrefixed.purs"),
);
