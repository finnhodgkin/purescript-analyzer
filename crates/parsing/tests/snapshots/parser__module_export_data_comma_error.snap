---
source: crates/parsing/tests/parser.rs
expression: "(node, errors)"
snapshot_kind: text
---
(
    Module@0..63
      ModuleHeader@0..63
        MODULE@0..6 "module"
        Comment@6..7
          WHITESPACE@6..7 " "
        ModuleName@7..34
          UPPER@7..33 "ModuleExportDataComma ..."
          Comment@33..34
            WHITESPACE@33..34 " "
        ModuleExportList@34..57
          LEFT_PARENTHESIS@34..35 "("
          ModuleExportType@35..55
            UPPER@35..41 "Either"
            ModuleExportTypeItemsList@41..55
              LEFT_PARENTHESIS@41..42 "("
              UPPER@42..46 "Left"
              COMMA@46..47 ","
              ERROR@47..48
                ERROR@47..47 ""
                COMMA@47..48 ","
              UPPER@48..53 "Right"
              ERROR@53..54
                ERROR@53..53 ""
                COMMA@53..54 ","
              RIGHT_PARENTHESIS@54..55 ")"
          RIGHT_PARENTHESIS@55..56 ")"
          Comment@56..57
            WHITESPACE@56..57 " "
        WHERE@57..62 "where"
        Comment@62..63
          WHITESPACE@62..63 "\n"
      LAYOUT_START@63..63 ""
      LAYOUT_END@63..63 ""
      END_OF_FILE@63..63 ""
    ,
    [
        ParseError {
            position: Position {
                offset: 47,
                line: 1,
                column: 48,
            },
            message: "Invalid token",
        },
        ParseError {
            position: Position {
                offset: 53,
                line: 1,
                column: 54,
            },
            message: "Trailing comma",
        },
    ],
)
